qiskit_optimization/algorithms/grover_optimizer.py

```diff
@@ -185,11 +139,8 @@ def solve(self, problem: QuadraticProgram) -> OptimizationResult:
             AttributeError: If the quantum instance has not been set.
             QiskitOptimizationError: If the problem is incompatible with the optimizer.
         """
-        if self._sampler is None and self._quantum_instance is None:
-            raise ValueError("A quantum instance or sampler must be provided.")
-
-        if self._quantum_instance is not None and self._sampler is not None:
-            raise ValueError("Only one of quantum_instance or sampler can be passed, not both!")
+        if self._sampler is None:
+            raise ValueError("A sampler must be provided.")
```

Comment:
This is just a comment on this. Given that a Sampler is now the only option with the removal of QI we could have defaulted the Sampler to the qiskit.primitives one if it was None in the constructor (and noted this for the sampler param there). That way the default value does not lead to an error - which seems strange. I get why it was done having to support both QI and Sampler. Since that is more of an improvement, and the GroverOptimizer was being looked at to be improved anyway, I think we can live with it like this, so as I said just a comment.

Reply:
Yes, we can make Sampler mandatory. But we need to change the argument order (need to move `sampler` first or second). Do you think whether we can change the argument order of `GroverOptimizer.__init__`?